examples/inference.py

```diff
@@ -0,0 +1,137 @@
+    if images_folder is None and dataframe_name is None:
+        raise InferenceConfigError("Either images_folder or dataframe_name should be set")
+
+    # Working with pure images if images folder is not None
+    if images_folder is not None:
+        extensions = (".jpeg", ".jpg", ".png", ".JPEG", ".JPG", ".PNG")
+        im_paths: List[Path] = []
+        for ext in extensions:
+            im_paths += images_folder.rglob("*" + ext)
+        bboxes = None
+
+    # Working with dataframe if it's path is not None
+    if dataframe_name is not None:
+        # read dataframe and get boxes. Use BaseDataset for inspiration
+        df = pd.read_csv(dataframe_name)
+        # begin check that columns are correct
+        is_path_column_ok = PATHS_COLUMN in df.columns
+        correct_column_names = (X1_COLUMN, X2_COLUMN, Y1_COLUMN, Y2_COLUMN)
+        coord_names_in_df = [coord in df.columns for coord in correct_column_names]
+        bboxes_exist = all(coord_names_in_df)
+        are_bad_boxes = not bboxes_exist and any(coord_names_in_df)
+        if not is_path_column_ok or are_bad_boxes:
+            bad_path_message = (
+                f"Path column not found. Please set it to {PATHS_COLUMN} " if not is_path_column_ok else ""
+            )
+            bad_columns_message = (
+                (
+                    f"Boxes are invalid: {df.columns}, please check that all of them exist and have "
+                    "correct names like this: {correct_column_names}"
+                )
+                if are_bad_boxes
+                else ""
+            )
+            raise InvalidDataFrameColumnsException(bad_path_message + bad_columns_message)
+        # end columns check
+        if bboxes_exist:
+            bboxes = []
+            for row in df[[X1_COLUMN, Y1_COLUMN, X2_COLUMN, Y2_COLUMN]].iterrows():
+                x1, y1, x2, y2 = row[1]
+                bboxes.append((x1, y1, x2, y2))
+        else:
+            bboxes = None
+
+        dataset_root = cfg["dataset_root"]
+        if dataset_root is not None:
+            dataset_root = Path(dataset_root)
+            df[PATHS_COLUMN] = df[PATHS_COLUMN].apply(lambda x: dataset_root / x)
+        else:
+            df[PATHS_COLUMN] = df[PATHS_COLUMN].apply(lambda x: Path(x))
+        im_paths = df[PATHS_COLUMN].tolist()
+
+        # Check that files from dataframe exist
+        for i, path in enumerate(im_paths):
+            if not path.exists():
+                raise FileExistsError(f"Could not find image on line {i+1}: {str(path)} in dataframe {dataframe_name}")
+
+    # Check that files could be opened
+    for path in im_paths:
+        with path.open("rb") as fimage:
+            if not verify_image_readable(fimage.read()):
+                raise InvalidImageException(f"While checking images encountered bad image: {path}")
+
+    kwargs = {}
+    # Loading transformations
+    trans_type = cfg.get("transforms")
```

Comment:
But I made is obligatory with the next lines

Reply:
But I see your point from this comment
https://github.com/OML-Team/open-metric-learning/pull/234/files#r1049430688

I'll make it a variable

Reply:
But code will work, because you use `.get` which return default `None`. As a result, `kwargs` is empty and you will use default transform of ListDataset

Reply:
Correct me if I'm wrong, but it should fail because on the line 123 I check if `trans_type` is not falsey value (`None` or empty string)
If it is falsey, I raise exception on the line 130